src/components/common/OfficeAddresses.vue

```diff
@@ -240,35 +253,14 @@ export default class OfficeAddresses extends Mixins(CommonMixin) {
    * This will be emitted back to the parent page when the addresses are updated.
    */
   @Prop({ default: null })
-  readonly addresses: BcorpAddressIf | null
+  readonly addresses: RegRecAddressesIF
 
-  /**
-   * Registered Office address object passed in from the parent which is pulled from store.
-   * This address is used as a baseline address in the event the user wants to reset the changes before submitting.
-   */
-  @Prop({ default: null })
-  private registeredAddress: BaseAddressObjIF
```

Comment:
The fundamental change here is that this component doesn't care what's in the store -- it will (re)load the office addresses for the specified As Of Date. (The COA filing is always "as of" today, but the AR filing can set an "as of" date.) (The API does not yet support returning office addresses "as of" a specified date but now the UI is ready for it.)

Reply:
very nice!